tests/test_utilities.py

```diff
@@ -968,3 +974,273 @@ def test_base_course_exporter_serialized_data_raises(self, mock_get_course_runs)
+
+    @mock.patch('integrated_channels.integrated_channel.learner_data.GradesApiClient')
+    @mock.patch('integrated_channels.integrated_channel.learner_data.CourseApiClient')
+    def test_learner_data_multiple_self_paced_courses(self, mock_course_api, mock_grades_api):
+        enrollment1 = EnterpriseCourseEnrollmentFactory(
+            enterprise_customer_user=self.enterprise_customer_user,
+            course_id=self.course_id,
+            consent_granted=True,
+        )
+        course_id2 = 'course-v1:edX+DemoX+DemoCourse2'
+        enrollment2 = EnterpriseCourseEnrollmentFactory(
+            enterprise_customer_user=self.enterprise_customer_user,
+            course_id=course_id2,
+            consent_granted=True,
+        )
+        enrollment3 = EnterpriseCourseEnrollmentFactory(
+            enterprise_customer_user=EnterpriseCustomerUserFactory(
+                user_id=UserFactory(username='R2D2').id,
+                enterprise_customer=self.enterprise_customer,
+            ),
+            course_id=self.course_id,
+            consent_granted=True,
+        )
+
+        def get_course_details(course_id):
+            """
+            Mock self-paced course details - set course_id to match input
+            """
+            return dict(
+                pacing='self',
+                course_id=course_id
+            )
+        mock_course_api.return_value.get_course_details.side_effect = get_course_details
+
+        def get_course_grade(course_id, username):
+            """
+            Mock grades data - set passed depending on course_id
+            """
+            return dict(
+                passed='2' in course_id,
+                course_key=course_id,
+                username=username,
+            )
+        mock_grades_api.return_value.get_course_grade.side_effect = get_course_grade
+
+        # Collect the learner data, with time set to NOW
+        with freeze_time(self.NOW):
+            learner_data = list(self.exporter.collect_learner_data())
+
+        assert len(learner_data) == 3
+        assert learner_data[0].enterprise_course_enrollment_id == enrollment1.id
+        assert learner_data[0].course_id == self.course_id
+        assert not learner_data[0].course_completed
+        assert learner_data[0].completed_timestamp is None
+        assert learner_data[0].grade == 'In Progress'
+
+        assert learner_data[1].enterprise_course_enrollment_id == enrollment3.id
+        assert learner_data[1].course_id == self.course_id
+        assert not learner_data[1].course_completed
+        assert learner_data[1].completed_timestamp is None
+        assert learner_data[1].grade == 'In Progress'
+
+        assert learner_data[2].enterprise_course_enrollment_id == enrollment2.id
+        assert learner_data[2].course_id == course_id2
+        assert learner_data[2].course_completed
+        assert learner_data[2].completed_timestamp == self.NOW_TIMESTAMP
+        assert learner_data[2].grade == 'Pass'
```

Comment:
@pomegranited Aren't pylint complaining about too long file? Probably it's a good point to split it up anyway.